overlord/snapstate/component.go

```diff
@@ -159,10 +154,16 @@ func componentSetupsForInstall(ctx context.Context, st *state.State, names []str
 	return componentTargetsFromActionResult("install", sars[0], names)
 }
 
-func installComponentAction(st *state.State, snapst SnapState, opts Options) (*store.SnapAction, error) {
-	si := snapst.CurrentSideInfo()
-	if si == nil {
-		return nil, errors.New("internal error: cannot install components for a snap that is not installed")
+func installComponentAction(st *state.State, snapst SnapState, snapRev snap.Revision, channel string, opts Options) (*store.SnapAction, error) {
+	var si *snap.SideInfo
+	if snapRev.Unset() {
+		si = snapst.CurrentSideInfo()
```

Comment:
mmh, test don't seem to hit this case?

Reply:
Looking at the code, we can't reach that case. Removed.